### ISSUE TITLE
Unable to resolve fully qualified hostnames on internal network after updating to 17.06

### ISSUE THREAD

**Issue by scottatron** [docker/for-mac#1811]
### Expected behavior

Before updating to 17.06, I was able to resolve hostnames on our internal domain (hq.local) using their fully qualified names (e.g. host.hq.local).

### Actual behavior

After updating to 17.06, I am unable to resolve hostnames on our internal domain using the their fully qualified names, but am able to resolve them using short names (e.g. ping host)

### Information

```
Docker for Mac: version: 17.06.0-ce-mac17 (4cdec4294a50b2233146b09469b49937dabdebdd)
macOS: version 10.12.5 (build: 16F73)
logs: /tmp/2EC7AE37-2E61-4BC8-8AFA-C446041EFD8A/20170703-125024.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

I have tried following the steps in `Library/Containers/com.docker.docker/Data/database/README` to use the experimental DNS resolver, but it did not change the behaviour.

I have also tried the workaround described in https://github.com/docker/for-mac/issues/1803#issuecomment-312372876 but it had no effect.

### Steps to reproduce the behavior

  1. Attach to VM with `docker run --rm -it --net host --pid host --privileged -v /:/rootfs alpine chroot /rootfs`
  2. Attempt to resolve a host using a fully qualified name on our internal domain: 

    curl -I host.hq.local
    curl: (6) Couldn't resolve host 'host.hq.local'

3. Attempt to resolve a host using a short hostname:
```
curl -I host
HTTP/1.1 200 OK
...
```

**Comment by scottatron** [comment]
@djs55 thanks for the speedy response on this issue. Will those vpnkit changes be likely to make it into the next edge build?

**Comment by djs55** [comment]
I've cherry-picked the fixes into the 17.06.1 branch, which should hopefully be released later this week. They'll also be in the next edge build.

Thanks again for your report!

**Comment by djs55** [comment]
The fix for this should be in the latest stable patch release:
```
Version 17.06.0-ce-mac19 (18663)
Channel: stable
c98c1c25e0
```
If there are still problems, could you upload a fresh set of diagnostics and let me know?

Thanks again for your report.
